Rework the Filecoin<->IPFS page (#477)
filecoin-project/filecoin-docs: docs/about-filecoin/ipfs-and-filecoin.md

# IPFS and Filecoin

Filecoin and IPFS are complementary systems for storing and sharing data in the distributed web. Both are free, open-source and share a number of building blocks, including data representation formats and certain protocols for network communication. However, the Filecoin and the IPFS networks are otherwise fully independent. Software that interacts with IPFS does not require Filecoin, and vice-versa, although some solutions may make use of both.

This page aims to explain the relationship between the IPFS and Filecoin and assist users in deciding which option, or combination of options, is best suited for their use-case.

## Data storage incentives

> [IPFS](https://ipfs.io) allows users to store and transfer verifiable, content-addressed data in a peer-to-peer network. IPFS users persist the data they want **on their own IPFS nodes**. This is referred as [pinning](https://docs.ipfs.io/concepts/persistence). Sometimes the data may be pinned using a third-party pinning service, or through groups of individual IPFS users. The data exists in the network as long as one user is storing it and able to provide it to others when they request it.

Author: There are subtleties here. `can provide it to others when requested` is ambiguous as it does not specify who requests it. It would seem as if the node storing it needs to request it.

I also don't fully understand why we remove all bold highlighting. There has to be a way to make certain key pieces of information stand out in a paragraph, it is not something we abuse.

However, IPFS does not include a built-in mechanism to incentivize the storage of data for _other_ people. This is the challenge Filecoin hopes to solve: the Filecoin network creates a distributed storage market place for long-term storage where nodes with large storage capacity can rent it to users with data-storage needs, and get paid for it.

> The cryptographic guarantees ensuring that data is safely stored by the Filecoin network make the storage and retrieval of data a computationally and time-expensive operation. For this reason, Filecoin enables an additional retrieval market where dedicated nodes can help quickly delivering content from the network for a payment. This delivery mechanism may make use of IPFS.

Author: This changes the meaning. The cryptography is not to ensure that data is retreivable. 

It is very different to say that "the crytography guarantees around storing and retrieving content are expensive" and that "the data is cryptographially guaranteed so that it is retrievable". It is nuanced but this editing departs from the original meaning.

> The cryptographic guarantees ensuring that data is safely stored by the Filecoin network make the storage and retrieval of data a computationally and time-expensive operation. For this reason, Filecoin enables an additional retrieval market where dedicated nodes can help quickly delivering content from the network for a payment. This delivery mechanism may make use of IPFS.

Reviewer: This is a _very_ wordy sentence that is difficult to follow:

> The cryptographic guarantees ensuring that data is safely stored by the Filecoin network make the storage and retrieval of data a computationally and time-expensive operation.

My commit was an attempt to make things easier to follow. If I've changed the meaning, however, can you try splitting the above sentence into two. That'd make this section clearer, without changing the meaning of the paragraph.

> The cryptographic guarantees ensuring that data is safely stored by the Filecoin network make the storage and retrieval of data a computationally and time-expensive operation. For this reason, Filecoin enables an additional retrieval market where dedicated nodes can help quickly delivering content from the network for a payment. This delivery mechanism may make use of IPFS.

Author: > This is a _very_ wordy sentence that is difficult to follow

I agree! I'll reword it tomorrow. Thanks for flagging!

In that sense, Filecoin can be seen as a "cold" storage layer, perfect to safely store large batches of data, while IPFS would be the "hot" storage layer, designed for the quick retrieval and distribution of content.

> ## Which should I use?

Author: Better systems. They are not single protocols. Protocol suites, if anything, and still it is complex because of how loaded the word "protocol" is in network architectures. Therefore I try to stay away from it.

### Using IPFS

- Data provided by user's own nodes. Otherwise must rely on other peers to voluntarily/altruistically storing data or on a centralized pinning service.
- Centralized IPFS pinning services must be trusted to do their job. IPFS brings no built-in provisions to verify that data is being stored and correctly provided by the pinning service.
- IPFS by itself works very well for popular content (with many providers), for organizations where there are incentives to sync and store data in multiple nodes and for situation where strong social contracts can be used to ensure the content remains hosted and maintained long-term.

### Using Filecoin

> - Clients make deals with miners to store data. The network verifies that the data is indeed stored by the miners, and regular, small payments are made for the duration of the deal.

Reviewer: ```suggestion
- Clients make _storage deals_ with miners to store data. The network verifies that the miners are correctly storing the data. Small payments are made on a regular basis for the duration of the _storage deal_.
```

> - Miners that are unable to honor the storage conditions they offered have to pay a penalty.

Reviewer: ```suggestion
- Miners that do not honor the storage deal are penalised.
```

> - Content retrieval might be offered by storage miners directly or by specialized retrieval miners, for a price.

Reviewer: ```suggestion
- Content retrieval might be offered by storage miners directly, or by specialized retrieval miners. The user requesting the data pays for this service.
```

> - Filecoin by itself work very well for large amounts of data to be stored for longer periods of time.

Reviewer: ```suggestion
- Filecoin excels at storing large amounts of data for long periods of time.
```

### Using both

> Some solutions like [Powergate](../build/powergate.md) combine the best of the two worlds, backing up data on the Filecoin network and at the same time providing it through the IPFS network. This ensures availability and fast retrieval while knowing, at the same time, that the data is safely backed up by one or several Filecoin miners.

Reviewer: ```suggestion
Some solutions combine the best of the to systems, backing up data on the Filecoin network and at the same time providing the data through the IPFS network. This ensures that data is constantly available and can be retrieved quickly, while also making sure that the data is safely backed up on the Filecoin network. [Powergate](../build/powergate.md) is a service that does this.
```

Author: s/to/two

Author: It sounds weird to say powergate is a service. It's a solution, or a system, or an application, or an integration, sometimes offered as a service.

Reviewer: The repo for Powergate reads:

> Powergate is a multitiered file storage API built on Filecoin and IPFS, and and index builder for Filecoin data. It's designed to be modular and extensible.

I'll just use that.

## The technology behind IPFS and Filecoin

Filecoin and IPFS share are powered by the same technology at many levels:

- [IPLD](https://ipld.io/) specifies data formats for content-addressed data like the blockchain or the way in which IPFS stores files.
- [libp2p](https://libp2p.io/) provides peer-to-peer network capabilities, connection security and key discovery and data distribution features like the DHT and Pubsub.
- [Multiformats](https://multiformats.io) define future-proof identifiers and data-types.
- [Graphsync](https://github.com/ipfs/go-graphsync) and [Bitswap](https://github.com/ipfs/go-bitswap) enable fast an efficient IPLD data transfers between nodes.

> **Interested in learning more?** Check some of our [application examples](../build/examples/README.md).

Reviewer: ```suggestion
Interested in learning more? Check some of our [application examples](../build/examples/README.md).
```
